api/triage_metadata.go

```diff
@@ -0,0 +1,231 @@
+	return tree, err
+}
+
+// createCommit creates the commit in the given reference using the given tree.
+func (tm triageMetadata) pushCommit(ref *github.Reference, tree *github.Tree) (err error) {
+	client := tm.githubClient
+	// Get the parent commit to attach the commit to.
+	parent, _, err := client.Repositories.GetCommit(tm.ctx, sourceOwner, sourceRepo, *ref.Object.SHA)
+	if err != nil {
+		return err
+	}
+	// This is not always populated, but is needed.
+	parent.Commit.SHA = parent.SHA
+
+	// Create the commit using the tree.
+	date := time.Now()
+	author := &github.CommitAuthor{Date: &date, Name: &tm.authorName, Email: &tm.authorEmail}
+	commit := &github.Commit{Author: author, Message: &commitMessage, Tree: tree, Parents: []github.Commit{*parent.Commit}}
+	newCommit, _, err := client.Git.CreateCommit(tm.ctx, sourceOwner, sourceRepo, commit)
+	if err != nil {
+		return err
+	}
+
+	// Attach the commit to the master branch.
+	ref.Object.SHA = newCommit.SHA
+	_, _, err = client.Git.UpdateRef(tm.ctx, sourceOwner, sourceRepo, ref, false)
+	return err
+}
+
+// createPR creates a pull request. Based on: https://godoc.org/github.com/google/go-github/github#example-PullRequestsService-Create
+func (tm triageMetadata) createPR(log shared.Logger) (err error) {
+	client := tm.githubClient
+	if prSubject == "" {
+		return errors.New("missing `-pr-title` flag; skipping PR creation")
+	}
+
+	newPR := &github.NewPullRequest{
+		Title:               &prSubject,
+		Head:                &commitBranch,
+		Base:                &prBranch,
+		Body:                &prDescription,
+		MaintainerCanModify: github.Bool(true),
+	}
+
+	pr, _, err := client.PullRequests.Create(tm.ctx, prRepoOwner, prRepo, newPR)
+	if err != nil {
+		return err
+	}
+
+	log.Infof("PR created: %s\n", pr.GetHTMLURL())
+	return nil
+}
+
+func (tm triageMetadata) mergeToGithub(triagedMetadataMap map[string][]byte, log shared.Logger) error {
+	ref, err := tm.getRef()
+	if err != nil {
+		log.Errorf("Unable to get/create the commit reference: %s\n", err)
+		return err
+	}
+
+	if ref == nil {
+		log.Errorf("No error returned but the reference is nil")
+		return errors.New("No error returned but the reference is nil")
+	}
+
+	tree, err := tm.getTree(ref, triagedMetadataMap)
+	if err != nil {
+		log.Errorf("Unable to create the tree based on the provided files: %s\n", err)
+		return err
+	}
+
+	if err := tm.pushCommit(ref, tree); err != nil {
+		log.Errorf("Unable to create the commit: %s\n", err)
+		return err
+	}
+
+	if err := tm.createPR(log); err != nil {
+		log.Errorf("Error while creating the pull request: %s", err)
+		return err
+	}
+
+	return nil
+}
+
+// Add Metadata into the existing Metadata YML files and only return modified files.
+func (tm triageMetadata) addToFiles(metadata shared.MetadataResults, filesMap map[string]shared.Metadata) map[string][]byte {
+	res := make(map[string][]byte)
+	for test, links := range metadata {
```

Comment:
Once I understood that this loop is *not* for filling `res`, I realized it would be useful to have a comment explaining what it *is* for. The overall purpose appears to be updating filesMap with the new information in metadata, so that we can then (afterwards) just walk the new metadata again and grab the YAML bytes from filesMap?

Reply:
Added!